File: agent-plugin/plugin/src/main/java/org/matilda/java/services/reflection/ObjectIdGenerator.java
package org.matilda.java.services.reflection;

import javax.inject.Inject;
import java.util.Random;

public class ObjectIdGenerator {
    @Inject
    Random mRandom;

    @Inject
    ObjectIdGenerator() {}


    public long generate(Object ignoredObject) {
        return mRandom.nextLong();
    }
}

File: agent-plugin/plugin/src/main/java/org/matilda/java/services/reflection/ObjectIdGenerator.kt
package org.matilda.java.services.reflection

import java.util.*
import javax.inject.Inject

class ObjectIdGenerator @Inject internal constructor() {
    @Inject
    lateinit var mRandom: Random
    fun generate(ignored: Any?): Long {
        return mRandom.nextLong()
    }
}
